internal/webhook/webhook.go

```diff
@@ -0,0 +1,193 @@
+// SPDX-FileCopyrightText: SAP SE or an SAP affiliate company and Gardener contributors
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package webhook
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/url"
+	"strings"
+
+	"github.com/go-jose/go-jose/v4"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+	certutil "k8s.io/client-go/util/cert"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/webhook"
+	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+
+	authenticationv1alpha1 "github.com/gardener/oidc-webhook-authenticator/apis/authentication/v1alpha1"
+)
+
+var (
+	// log is for logging in this package.
+	log = logf.Log.WithName("openidconnect-resource")
+
+	allowedSigningAlgs = map[authenticationv1alpha1.SigningAlgorithm]bool{
+		authenticationv1alpha1.RS256: true,
+		authenticationv1alpha1.RS384: true,
+		authenticationv1alpha1.RS512: true,
+		authenticationv1alpha1.ES256: true,
+		authenticationv1alpha1.ES384: true,
+		authenticationv1alpha1.ES512: true,
+		authenticationv1alpha1.PS256: true,
+		authenticationv1alpha1.PS384: true,
+		authenticationv1alpha1.PS512: true,
+	}
+)
+
+// +kubebuilder:webhook:path=/webhooks/mutating,mutating=true,failurePolicy=fail,sideEffects=None,groups=authentication.gardener.cloud,resources=openidconnects,verbs=create;update,versions=v1alpha1,name=oidc.authentication.gardener.cloud,admissionReviewVersions={v1,v1beta1}
+// +kubebuilder:webhook:verbs=create;update,path=/webhooks/validating,mutating=false,failurePolicy=fail,sideEffects=None,groups=authentication.gardener.cloud,resources=openidconnects,versions=v1alpha1,name=oidc.authentication.gardener.cloud,admissionReviewVersions={v1,v1beta1}
+
+// Handler is a webhook implementing defaulting and validation for the OpenIDConnect resource.
+type Handler struct{}
+
+// ensure webhookHandler implements CustomDefaulter and CustomValidator interfaces
+var _ webhook.CustomDefaulter = (*Handler)(nil)
+var _ webhook.CustomValidator = (*Handler)(nil)
+
+// Default implements webhook.CustomDefaulter so a webhook will be registered for the type
+func (Handler) Default(_ context.Context, obj runtime.Object) error {
```

Comment:
Should we use pointer receivers?

```suggestion
func (*Handler) Default(_ context.Context, obj runtime.Object) error {
```

Reply:
We can, then we just have to use pointer to register the handler
```go
webhookHandler := internalwebhook.Handler{}
defaultingWebhook := admission.WithCustomDefaulter(scheme, oidc, &webhookHandler)
```

Do you see any advantage/disadvantage of using pointer vs. value?